CUDADataFormats/SiPixelDigi/interface/SiPixelDigisCUDASOAView.h

```diff
@@ -0,0 +1,48 @@
+#ifndef CUDADataFormats_SiPixelDigi_interface_SiPixelDigisCUDASOAView_h
+#define CUDADataFormats_SiPixelDigi_interface_SiPixelDigisCUDASOAView_h
+
+#include <cuda_runtime.h>
+
+#include "HeterogeneousCore/CUDAUtilities/interface/device_unique_ptr.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/host_unique_ptr.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/cudaCompat.h"
+
+class SiPixelDigisCUDASOAView {
+public:
+  enum class StorageLocation { XX = 0, YY = 1, MODULEIND = 2, ADC = 3, CLUS = 4, PDIGI = 6, RAWIDARR = 8, MAX = 10 };
+  /*
+  ============================================================================================================================
+  |    XX     |     YY    | MODULEIND |    ADC    |          CLUS         |          PDIGI         |         RAWIDARR        |
+  ============================================================================================================================
+  |  0: N*16  |  1: N*16  |  2: N*16  |  3: N*16  |        4: N*32        |         6: N*32        |         8: N*32         |
+  ============================================================================================================================
+  */
+  // These are for CPU output
+  // we don't copy local x and y coordinates and module index
+  enum class StorageLocationHost { ADC = 0, CLUS = 1, PDIGI = 3, RAWIDARR = 5, MAX = 7 };
+  /*
+  ========================================================================================
```

Comment:
what happens if N is odd? (the  32bit word would not be alligned...)
I would feel more comfortable reversing the whole storage
so we keep
clus,adc,id,yy,xx close to each other
and one transfer to host the first part (that is also much better aligned)

Reply:
Is the 128-byte alignment not important for each column then?

Reply:
it should be

Reply:
Looks like the simpler fix would be to round `maxFedWords` up the next multiple of `64` in the constructor of `SiPixelDigisCUDA`.

This would guarantee that all "columns" are aligned to 128 bytes (assuming the whole buffer is) and that any 32-bit integer does not fall on a 16-bit address.